src/components/inputs/localized-money-input/localized-money-input.js

```diff
@@ -0,0 +1,307 @@
+
+  static propTypes = {
+    id: PropTypes.string,
+    name: PropTypes.string,
+    value: PropTypes.shape({
+      amount: PropTypes.string.isRequired,
+      currencyCode: PropTypes.string.isRequired,
+    }).isRequired,
+    onChange: PropTypes.func,
+    currency: PropTypes.string.isRequired,
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    isDisabled: PropTypes.bool,
+    currenciesControl: PropTypes.node,
+    hasError: PropTypes.bool,
+    hasWarning: PropTypes.bool,
+    placeholder: PropTypes.string,
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+    error: PropTypes.node,
+    warning: PropTypes.node,
+  };
+
+  handleChange = event => {
+    // We manipulate the event to add the currency to the target.
+    // That way the users  can read
+    // event.target.currency and event.target.value to determine the next value.
+    //
+    // We only need this information for the story, the MC application code will
+    // never need to access the information in such an inconvenient way, as
+    // Formik can deal with a name like "foo.en" and sets the value correctly.
+    // We can't use this as we aren't guaranteed a name in the story as the user
+    // might clear it using the knob, and then we can't parse the currency from
+    // the input name anymore.
+    //
+    // eslint-disable-next-line no-param-reassign
+    event.target.currency = this.props.currency;
+    this.props.onChange(event);
+  };
+
+  render() {
+    return (
+      <Spacings.Stack scale="xs">
+        <div className={styles.fieldContainer}>
+          <MoneyInput
+            id={this.props.id}
+            name={this.props.name}
+            value={this.props.value}
+            onChange={this.handleChange}
+            onBlur={this.props.onBlur}
+            isDisabled={this.props.isDisabled}
+            placeholder={this.props.placeholder}
+            hasError={this.props.hasError}
+            hasWarning={this.props.hasWarning}
+            {...filterDataAttributes(this.props)}
+          />
+        </div>
+        <div className={styles.commandsContainer}>
+          <div className={styles.commandsLeft}>
+            {do {
+              if (this.props.error) <div>{this.props.error}</div>;
+              else if (this.props.warning) <div>{this.props.warning}</div>;
+              else this.props.currenciesControl;
+            }}
+          </div>
+        </div>
+        {(this.props.error || this.props.warning) &&
+          this.props.currenciesControl}
+      </Spacings.Stack>
+    );
+  }
+}
+
+export class LocalizedMoneyInput extends React.Component {
+  static displayName = 'LocalizedMoneyInput';
+
+  static propTypes = {
+    id: PropTypes.string,
+    name: PropTypes.string,
+    // then input doesn't accept a "currencies" prop, instead all possible
+    // currencies have to exist (with empty or filled strings) on the value:
+    //   { EUR: 12.00, USD: 14.23, CAD: 13.70 }
```

Comment:
The example given here doesn't use strings but actual numbers. However the prop-type requires strings to be present

```suggestion
    //   { EUR: "12.00", USD: "14.23", CAD: "" }
```